--- lib/Lua/loslib.c	2012-01-17 20:37:27.796802580 +0100
+++ lib/Lua/loslib.c	2012-01-17 20:37:12.606736469 +0100
@@ -262,12 +262,6 @@
 }
 
 
-static int os_difftime (lua_State *L) {
-  lua_pushnumber(L, difftime((time_t)(luaL_checknumber(L, 1)),
-                             (time_t)(luaL_optnumber(L, 2, 0))));
-  return 1;
-}
-
 /* }====================================================== */
 
 
@@ -299,7 +293,6 @@
 static const luaL_Reg syslib[] = {
   {"clock",     os_clock},
   {"date",      os_date},
-  {"difftime",  os_difftime},
   {"execute",   os_execute},
   {"exit",      os_exit},
   {"getenv",    os_getenv},
